force-app/main/educationCloudSettings/classes/settingsProduct/EDCSettingsProductAPIService_TEST.cls

```diff
@@ -122,6 +122,38 @@ private class EDCSettingsProductAPIService_TEST {
         System.assertEquals(icon, product.icon, 'icon should be set to ' + icon);
     }
 
+    /**************************************************************************************************************************
+     * @description Test method to verify a EDCSettingsProductApiService.SettingsInvalidResponseException is thrown when an
+     * invalid JSON response is received
+     ***************************************************************************************************************************/
+    @isTest
+    private static void testExceptionGetSettingsProductInformationBadResponse() {
+        String classname = 'EDASettingsProductInformationAPIService';
+        String namespace = 'hed';
+        Decimal apiVersion = 52.0;
+        String exceptionMessage;
+        EDCSettingsProdInfoActionResultModel actionResult;
+        EDCSettingsProductAPIService.instance = new EDCSettingsProductAPIServiceBadResponseStub();
+
+        try {
+            Test.startTest();
+            actionResult = EDCSettingsProductAPIService.getInstance()
+                .getSettingsProductInformation(classname, namespace, apiVersion);
+            Test.stopTest();
+            System.assert(true, 'this should not happen');
```

Comment:
I believe it should be `System.assert(false, ...` here and below 
```suggestion
            System.assert(false, 'this should not happen');
```

Reply:
@iskander-sfdc I added that assert to make sure it fails if we don't receive an exception, basically the try should not finish executing until that line because I want it to catch the exception as part of this test scenario.

Reply:
Thanks, just wanted to make sure it works in namespaced org
Maybe we can test in a namespaced scratch org - `cci flow run dev_org_namespaced`

Reply:
@salvadorsfdc `System.assert(true, 'this should not happen');` doesn't do anything
`System.assert(false, 'this should not happen');` throws assert exception. I believe this is what you wanted to make sure the exception is thrown at this line, right?

Reply:
oh you are right, it needs to be false in order to fail, thanks!!!

Reply:
Done. Thanks!